sale_commercial_partner/models/sale.py

```diff
@@ -12,5 +12,6 @@ class SaleOrder(models.Model):
         related='partner_id.commercial_partner_id',
         store=True,
         string='Commercial Entity',
-        index=True
+        compute_sudo=True,
```

Comment:
related_sudo can be used instead, If you want:
https://github.com/odoo/odoo/blob/7b986587956c23b77b0da5043410eab11e0d2f28/odoo/fields.py#L368

Reply:
The linked code is for v13. It looks to be different in v12, where that evaluation isn't made